langchain4j-vespa/src/main/java/dev/langchain4j/store/embedding/vespa/VespaEmbeddingStoreImpl.java

```diff
@@ -0,0 +1,264 @@
+    this.rankProfile = rankProfile != null ? rankProfile : DEFAULT_RANK_PROFILE;
+    this.targetHits = targetHits != null ? targetHits : DEFAULT_TARGET_HITS;
+    this.avoidDups = avoidDups != null ? avoidDups : DEFAULT_AVOID_DUPS;
+  }
+
+  @Override
+  public String add(Embedding embedding) {
+    return add(null, embedding, null);
+  }
+
+  @Override
+  public void add(String id, Embedding embedding) {
+    add(id, embedding, null);
+  }
+
+  @Override
+  public String add(Embedding embedding, TextSegment textSegment) {
+    return add(null, embedding, textSegment);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings) {
+    return addAll(embeddings, null);
+  }
+
+  @Override
+  public List<String> addAll(List<Embedding> embeddings, List<TextSegment> embedded) {
+    if (embedded != null && embeddings.size() != embedded.size()) {
+      throw new IllegalArgumentException("The list of embeddings and embedded must have the same size");
+    }
+
+    List<String> ids = new ArrayList<>();
+
+    try (JsonFeeder jsonFeeder = buildJsonFeeder()) {
+      List<Record> records = new ArrayList<>();
+
+      for (int i = 0; i < embeddings.size(); i++) {
+        records.add(buildRecord(embeddings.get(i), embedded != null ? embedded.get(i) : null));
+      }
+
+      jsonFeeder.feedMany(
+        Json.toInputStream(records, List.class),
+        new JsonFeeder.ResultCallback() {
+          @Override
+          public void onNextResult(Result result, FeedException error) {
+            if (error != null) {
+              throw new RuntimeException(error.getMessage());
+            } else if (Result.Type.success.equals(result.type())) {
+              ids.add(result.documentId().toString());
+            }
+          }
+
+          @Override
+          public void onError(FeedException error) {
+            throw new RuntimeException(error.getMessage());
+          }
+        }
+      );
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    return ids;
+  }
+
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults) {
+    return findRelevant(referenceEmbedding, maxResults, DEFAULT_MIN_SCORE);
+  }
+
+  @Override
+  public List<EmbeddingMatch<TextSegment>> findRelevant(Embedding referenceEmbedding, int maxResults, double minScore) {
+    try (CloseableHttpClient httpClient = buildQueryClient()) {
+      String searchQuery = Q
+        .select(FIELD_NAME_DOCUMENT_ID, FIELD_NAME_TEXT_SEGMENT, FIELD_NAME_VECTOR)
+        .from(documentType)
+        //              .where(Q.nearestNeighbor(FIELD_NAME_VECTOR, "q").annotate(A.a("targetHits", targetHits)))
+        .where(buildNearestNeighbor())
+        .fix()
+        .hits(maxResults)
+        .ranking(rankProfile)
+        .param("input.query(q)", Json.toJson(referenceEmbedding.vectorAsList()))
+        .param("input.query(threshold)", String.valueOf(minScore))
```

Comment:
here we need String.valueOf((minScore * 2) - 1)

Reply:
same here